pkgs/development/python-modules/tpm2-pytss/default.nix

```
@@ -32,10 +36,22 @@ buildPythonPackage rec {
   patches = [
     # Fix hardcoded `fapi-config.json` configuration path
     ./fapi-config.patch
+  ] ++ lib.optionals isCross [
+    # pytss will regenerate files from headers of tpm2-tss.
+    # Those headers are fed through a compiler via pycparser. pycparser expects `cpp`
+    # to be in the path.
+    # This is put in the path via stdenv when not cross-compiling, but this is absent
+    # when cross-compiling is turned on.
+    # This patch changes the call to pycparser.preprocess_file to provide the name
+    # of the cross-compiling cpp
+    ./cross.patch
```

```suggestion
    (substituteAll {
      src = ./cross.patch;
      CROSS_PREFIX = stdenv.hostPlatform.config;
    })
```

No need to `sed` in this way (try this yourself first)

---

thanks!